dali/util/shared_mem.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_UTIL_SHARED_MEM_H_
+#define DALI_UTIL_SHARED_MEM_H_
+
+#include <stdint.h>
+#include <memory>
+#include "dali/core/common.h"
+
+namespace dali {
+namespace python {
+
+class DLL_PUBLIC ShmFdWrapper {
+ public:
+  int fd;
+
+  explicit ShmFdWrapper(int fd);
+
+  ShmFdWrapper();
+
+  ~ShmFdWrapper();
+
+  DISABLE_COPY_MOVE_ASSIGN(ShmFdWrapper);
+
+  int close();
+};
+
+class DLL_PUBLIC MapMemWrapper {
+  uint64_t size_;
+
+ public:
+  uint8_t *ptr;
+
+  MapMemWrapper(int fd, uint64_t size);
+
+  ~MapMemWrapper();
+
+  DISABLE_COPY_MOVE_ASSIGN(MapMemWrapper);
+
+  int unmap();
+};
+
+class DLL_PUBLIC SharedMem {
+ private:
```

Comment:
I would move the public interface to the beginning, and private fields to the end.